lib/sendf.c

```diff
@@ -406,7 +406,9 @@ CURLcode Curl_write_plain(struct Curl_easy *data,
 {
   CURLcode result;
   struct connectdata *conn = data->conn;
-  int num = (sockfd == conn->sock[SECONDARYSOCKET]);
+  int num;
+  DEBUGASSERT(conn);
+  num = (sockfd == conn->sock[SECONDARYSOCKET]);
```

Comment:
By the way, while you are currently refactoring this, wouldn't it be even better to use a bool for this and not a 4-byte huge integer?

Reply:
`int` is the "native" variable type for the CPU which traditionally and frequently is faster and better to use than any other "custom" size variable. Forcing it to something else might (only *might*, because we don't know what "bool" will actually use) take a few bytes less stack but can in worst case require more instructions.

I don't think its worth experimenting with this just to *possibly* gain 3 bytes of stack use.